packages/@aws-cdk/aws-appmesh/lib/route.ts

```diff
@@ -111,21 +116,37 @@ export interface RouteProps extends RouteBaseProps {
  */
 export class Route extends cdk.Resource implements IRoute {
   /**
-   * Import an existing route given an ARN
+   * Import an existing Route given an ARN
    */
   public static fromRouteArn(scope: Construct, id: string, routeArn: string): IRoute {
-    return new ImportedRoute(scope, id, { routeArn });
+    return new class extends cdk.Resource implements IRoute {
+      readonly routeArn = routeArn;
+      readonly mesh = Mesh.fromMeshName(this, 'Mesh', cdk.Fn.select(0, cdk.Fn.split('/', cdk.Stack.of(scope).parseArn(routeArn).resourceName!)));
+      readonly virtualRouter = VirtualRouter.fromVirtualRouterAttributes(this, 'VirtualRouter', {
+        mesh: this.mesh,
+        virtualRouterName: cdk.Fn.select(2, cdk.Fn.split('/', cdk.Stack.of(scope).parseArn(routeArn).resourceName!)),
+      });
+      readonly routeName = cdk.Fn.select(4, cdk.Fn.split('/', cdk.Stack.of(scope).parseArn(routeArn).resourceName!));
+    }(scope, id);
```

Comment:
The expression `cdk.Fn.split('/', cdk.Stack.of(scope).parseArn(routeArn).resourceName!)` is repeated like 5 times here... can you extract it to a local variable instead?

Reply:
Yeah, would be more readable for sure, good call